packages/contracts-bedrock/scripts/hash-onion/GenerateHashOnion.s.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import { Script } from "forge-std/Script.sol";
+import { console } from "forge-std/Console.sol";
+
+/// @notice Script to generate the hash onion from the tokens json file.
+///         The tokens json file needs to be created on the path specified in the `_tokensPath` function. The keys of
+///         the json must be `localToken` and `remoteToken` with the addresses of the tokens. The
+///         `test-mock-tokens.json` file can be used as a reference (it's only for testing purposes).
+contract GenerateHashOnion is Script {
+    /// @notice Struct to hold the local and remote token addresses.
+    /// @param localToken  Address of the local token.
+    /// @param remoteToken Address of the remote token.
+    struct TokenPair {
+        address localToken;
+        address remoteToken;
+    }
+
+    /// @notice Initial layer of the hash onion.
+    /// keccak256(abi.encode(0));
+    bytes32 internal constant INITIAL_ONION_LAYER = 0x290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563;
+
+    /// @notice Generates the hash onion from the tokens json file.
+    /// @return _hashOnion The hash onion value.
+    function run() public view virtual returns (bytes32 _hashOnion) {
+        // Read the json of the tokens and parse it
+        TokenPair[] memory _tokensPairs = _parseJson(_tokensPath());
+
+        // Generate the hash onion and print it
+        _hashOnion = _generateHashOnion(_tokensPairs);
+        console.log("Generated hash onion: ");
+        console.logBytes32(_hashOnion);
+    }
+
+    /// @notice Helper function to get the path of the tokens json file.
+    ///         Forge must have read permissions allowed to the path.
+    /// @return _path Path of the tokens json file.
+    function _tokensPath() internal view virtual returns (string memory _path) {
+        _path = string.concat(vm.projectRoot(), "/scripts/hash-onion/tokens.json");
```

Comment:
I was in doubt if hardcoding it or getting that value from the `.env`.
I opted for this way since it's easier and more straightforward than creating an env file just for this case - also taking into account that is highly likely this script will only be used once per chain that needs to update their deployments mapping.